src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -615,30 +467,30 @@ public SolutionState RemoveProject(ProjectId projectId)
         /// Creates a new solution instance with the project specified updated to have the new
         /// assembly name.
         /// </summary>
-        public SolutionState WithProjectAssemblyName(ProjectId projectId, string assemblyName)
+        public (SolutionState, ProjectState) WithProjectAssemblyName(ProjectId projectId, string assemblyName)
```

Comment:
to aid SolutionCompilationState, a lot of these methods return not just the update solution, but also teh project that was updated.  That way SolutionCompilationState can easily then pass those into its own forking code directly.